sqlglot/generator.py

```diff
@@ -3518,6 +3518,9 @@ def sub_sql(self, expression: exp.Sub) -> str:
     def trycast_sql(self, expression: exp.TryCast) -> str:
         return self.cast_sql(expression, safe_prefix="TRY_")
 
+    def jsoncast_sql(self, expression: exp.JSONCast) -> str:
+        return self.cast_sql(expression)
```

Comment:
I put this here to allow a best-effort transpilation. I'm not sure, but perhaps this could allow CH ~> Snowflake. The latter has a `VARIANT` type that kind of matches `Dynamic`.

Reply:
CH has both Variant and Dynamic. :)
Variant has a strict list of types. While Dynamic just accepts anything and adds it to the list.